operator/v1/types_console.go

```diff
@@ -143,12 +143,51 @@ type ProjectAccess struct {
 	AvailableClusterRoles []string `json:"availableClusterRoles,omitempty"`
 }
 
+// CatalogTypesState defines the state of the catalog types based on which the types will be enabled or disabled.
+type CatalogTypesState string
+
+const (
+	Enabled  CatalogTypesState = "Enabled"
+	Disabled CatalogTypesState = "Disabled"
+)
+
+// DeveloperConsoleCatalogTypesState defines the state of the sub-catalog types.
+// +union
+type DeveloperConsoleCatalogTypesState struct {
```

Comment:
Based on the testing I'm doing here you may need a couple of changes to make sure this works. You have a required field within an optional struct, we need to make sure it's omitted correctly.

https://github.com/openshift/api/blob/b4c38862d26f227af4221d01dd41331e2b78201a/example/v1/types_stable.go#L70-L87

I think you need `omitempty` for state, and then we can add the CEL as in that example for the enabled and disabled fields here